src/bundles.ts

```diff
@@ -46,10 +51,9 @@ async function* prepareRemoteBundle({
 }: PrepareRemoteBundleOptions): AsyncGenerator<Result<RemoteBundle, BundleErrorCodes>> {
   let response: Result<RemoteBundle, BundleErrorCodes>;
   let { bundleHash } = options;
-
-  const fileChunks = chunk(options.files, maxPayload / 300);
-  emitter.createBundleProgress(0, fileChunks.length);
-  for (const [i, chunkedFiles] of fileChunks.entries()) {
+  let cumulativeProgress = 0;
+  emitter.createBundleProgress(cumulativeProgress, options.files.length);
+  for (const chunkedFiles of composeFilePayloads(options.files, maxPayload)) {
```

Comment:
Have we considered using a different loop than `for...of`? If we're talking optimizations, this is the slowest one...

Reply:
Do you have a specific suggestion?

Reply:
You take a look at this:
 https://link.medium.com/Zr7kdVR0djb

tl;dr
If possible, the plain old `for` would be the fastest